File: src/main/java/repository/ITripRepository.java
package repository;

import model.Trip;
import java.util.List;

public interface ITripRepository {
    void save(Trip trip);
    void update(Trip trip);
    void delete(int id);
    Trip findById(int id);
    List<Trip> findAll();
    List<Trip> findByAttractionAndDepartureInterval(String attraction, String startTime, String endTime);
    void updateAvailableSeats(int tripId, int newAvailableSeats);
}

File: TurismUI/TurismClient/Repositories/ITripRepository.cs
using System.Collections.Generic;
using TurismClient.Models;

namespace TurismClient.Repositories;

public interface ITripRepository
{
    void Save(Trip trip);
    Trip? FindById(int id);
    IEnumerable<Trip> FindAll();
    IEnumerable<Trip> FindByAttractionAndDepartureInterval(string attraction, string startDate, string endDate);
    void UpdateAvailableSeats(int tripId, int newAvailableSeats);
}